emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -0,0 +1,311 @@
+                    .map(this::parseViolation)
+                    .collect(Collectors.toSet());
+        } catch (IOException exception) {
+            return new HashSet<>();
+        }
+    }
+
+    /**
+     * Scrubs newViolations of violations believed to be duplicates from violation-counts-old.
+     */
+    private void removeDuplicateViolations() {
+        Set<List<String>> violationsToRemove = new HashSet<>();
+        for (List<String> newViolation : newViolations) {
+            Set<List<String>> relevantOldViolations = oldViolations.stream()
+                    .filter(oldViolation -> oldViolation.get(0).equals(newViolation.get(0))) // same spec
+                    .filter(oldViolation -> oldViolation.get(1).equals(newViolation.get(1))
+                                         || isRenamed(oldViolation.get(1), newViolation.get(1))) // same class
+                    .filter(oldViolation -> hasSameLineNumber(oldViolation.get(1), Integer.parseInt(oldViolation.get(2)),
+                            Integer.parseInt(newViolation.get(2)))) // same line number
+                    .collect(Collectors.toSet());
+
+            if (!relevantOldViolations.isEmpty()) {
+                violationsToRemove.add(newViolation);
+            }
+        }
+        for (List<String> violationToRemove : violationsToRemove) {
+            newViolations.remove(violationToRemove);
+        }
+    }
+
+    /**
+     * Determines whether an old class has been renamed to the new one or not.
+     *
+     * @param oldClass Previous possible name of a class
+     * @param newClass Rename being considered
+     * @return Whether the old class name was renamed to the new one
+     */
+    private boolean isRenamed(String oldClass, String newClass) {
+        for (String renamedClass : renames.keySet()) {
+            if (renamedClass.contains(newClass) && renames.get(renamedClass).contains(oldClass)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether an old line in a file can be mapped to the new line. There's room for optimization.
+     *
+     * @param classInfo Particular class being considered (if the class was renamed, this is the old name)
+     * @param oldLine Original line number
+     * @param newLine New line number
+     * @return Whether the original line number can be mapped to the new line number in the updated version
+     */
+    private boolean hasSameLineNumber(String classInfo, int oldLine, int newLine) {
+        for (String className : lineChanges.keySet()) {
+            if (className.contains(classInfo)) {
+                int offset = 0;
+                for (Integer originalLine : lineChanges.get(className).keySet()) {
+                    if (originalLine <= oldLine) {
+                        offset += lineChanges.get(className).get(originalLine);
+                    }
+                }
+                if (newLine - oldLine <= offset) {
```

Comment:
It's likely that I'm misunderstanding how this works, but I'm skeptical of this line. If all modifications were in place, then `offset` should be 0. In which case, we'd be returning `true` whenever `newLine <= oldLine`, which  doesn't seem right.

Reply:
Yeah this only works for insertions, I updated the logic to account for both insertions and deletions